keras_applications/imagenet_utils.py

```diff
@@ -113,6 +113,9 @@ def _preprocess_symbolic_input(x, data_format, mode, **kwargs):
 
     backend, _, _, _ = get_submodules_from_kwargs(kwargs)
 
+    if backend.dtype(x) != backend.floatx():
+        x = backend.cast(x, backend.floatx())
```

Comment:
Trying to replicate the logic as in:
https://github.com/keras-team/keras-applications/pull/129/files#diff-b06113767301a716ab58d22bd182226aL39-L40